python/core/qgsvector.sip

```diff
@@ -1,47 +1,163 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/core/qgsvector.h                                                 *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
 class QgsVector
 {
-%TypeHeaderCode
-#include <qgsvector.h>
+%Docstring
+ A class to represent a vector.
+ Currently no Z axis / 2.5D support is implemented.
 %End
 
+%TypeHeaderCode
+#include "qgsvector.h"
+%End
   public:
+
     QgsVector();
+%Docstring
+ Default constructor for QgsVector. Creates a vector with length of 0.0.
+%End
 
     QgsVector( double x, double y );
+%Docstring
+ Constructor for QgsVector taking x and y component values.
+ \param x x-component
+ \param y y-component
+%End
 
     QgsVector operator-() const;
+%Docstring
+Swaps the sign of the x and y components of the vector.
+ :rtype: QgsVector
+%End
 
     QgsVector operator*( double scalar ) const;
+%Docstring
+ Returns a vector where the components have been multiplied by a scalar value.
+ \param scalar factor to multiply by
+ :rtype: QgsVector
+%End
 
     QgsVector operator/( double scalar ) const;
+%Docstring
+ Returns a vector where the components have been divided by a scalar value.
+ \param scalar factor to divide by
+ :rtype: QgsVector
+%End
 
     double operator*( QgsVector v ) const;
+%Docstring
+ Returns the dot product of two vectors, which is the sum of the x component
+  of this vector multiplied by the x component of another
+  vector plus the y component of this vector multiplied by the y component of another vector.
+ :rtype: float
+%End
 
     QgsVector operator+( QgsVector other ) const;
+%Docstring
+ Adds another vector to this vector.
+.. versionadded:: 3.0
+ :rtype: QgsVector
+%End
 
-    QgsVector& operator+=( QgsVector other );
+    QgsVector &operator+=( QgsVector other );
+%Docstring
+ Adds another vector to this vector in place.
+.. versionadded:: 3.0
+ :rtype: QgsVector
+%End
 
     QgsVector operator-( QgsVector other ) const;
```

Comment:
Not strictly required, but can you change the method to

QgsVector operator-( const QgsVector &other );